manifests/pin.pp

```diff
@@ -65,8 +63,8 @@
   $file_name = regsubst($title, '[^0-9a-z\-_\.]', '_', 'IG')
 
   $path = $order ? {
-    ''      => "${preferences_d}/${file_name}.pref",
-    default => "${preferences_d}/${order}-${file_name}.pref",
+    ''      => "${apt::params::preferences_d}/${file_name}.pref",
```

Comment:
Couldn't his just be `apt::preferences_d`? We require `apt` and it inherits from `params`.t